pkg/controllers/gateway/params.go

```diff
@@ -0,0 +1,89 @@
+package gateway
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+type Params struct {
+}
+
+var defaultParams Params = Params{}
+
+type InvalidParamsError struct {
+	message string
+}
+
+var _ error = &InvalidParamsError{}
+
+// Error implements error
+func (e *InvalidParamsError) Error() string {
+	return e.message
+}
+
+func IsInvalidParamsError(err error) (is bool) {
+	_, is = err.(*InvalidParamsError)
+	return
+}
+
+type ParamsResolver func(context.Context, client.Client, gatewayv1beta1.ParametersReference) (*Params, error)
+
+var paramsResolvers map[schema.GroupKind]ParamsResolver = map[schema.GroupKind]ParamsResolver{}
+
+func fromConfigMap(ctx context.Context, client client.Client, paramsRef gatewayv1beta1.ParametersReference) (*Params, error) {
+	if paramsRef.Namespace == nil || *paramsRef.Namespace == "" {
+		return nil, &InvalidParamsError{"Namespace must be defined for ConfigMap parameters"}
+	}
+
+	configMap := &corev1.ConfigMap{}
+	namespace := string(*paramsRef.Namespace)
+
+	if err := client.Get(ctx, types.NamespacedName{
+		Name:      paramsRef.Name,
+		Namespace: namespace,
+	}, configMap); err != nil {
+		return nil, &InvalidParamsError{fmt.Sprintf("failed to get config map %s/%s: %s", namespace, paramsRef.Name, err.Error())}
+	}
+
+	// TODO: parse configMap data into Params
+	result := &Params{}
+	return result, nil
+}
+
+func init() {
+	configMapGroupKind := schema.GroupKind{
+		Group: corev1.GroupName,
+		Kind:  "ConfigMap",
+	}
+	paramsResolvers[configMapGroupKind] = fromConfigMap
```

Comment:
I wonder if we need this, I think we are only ever likely to have one params object type

Reply:
The motivation for it is to be able to support multiple types in case we want to for example define a CRD for the parameters. So the user can have the option to define the parameters in a ConfigMap or in a CR and the controller will be able to handle it